File: contest/icpc/jag/2014/a.cpp
#include <bits/stdc++.h>
#define rep(i,n) for(int i = 0; i < (int)(n); i++)
#define REP(i,k,n) for(int i = k; i < (int)(n); i++)

#define vi vector<int>
#define pb push_back

typedef long long ll;
using namespace std;



int main(){
    int n;
    while(cin >> n,n) {
        int up = 0,down = 0;
        int cnt = 0;
        rep(i,n){
            string tmp;
            getline(cin, tmp, ' ');
            cout << i << tmp << endl;
            if(tmp == "lu" || tmp == "ru") up++, down = 0;
            else if(tmp == "ld" || tmp == "rd") down++, up = 0;
            if(up == 2 || down == 2) cnt++;
        }
        cout << cnt << endl;
    }
    return 0;
}

// getline の挙動がおかしい

File: contest/icpc/jag/2014/a.py
while(True):
    n = input()
    if n == '0': break
    line = input()
    qs = line.split(" ")
    up = 0
    down = 0
    cnt = 0
    for q in qs:
        if q == "lu" or q == "ru" :
            up += 1
            down = 0
        else:
            down += 1
            up = 0
        if up==2 or down == 2:
            cnt += 1
    print(cnt)
